drivers/SmartThings/sonos/src/api/sonos_connection.lua

```diff
@@ -329,11 +368,14 @@ function SonosConnection.new(driver, device)
         end
       end
     else
-      log.warn(string.format("WebSocket Message for %s did not have a data payload: %s", device_name, st_utils.stringify_table(msg)))
+      log.warn(string.format("WebSocket Message for %s did not have a data payload: %s", device_name,
+        st_utils.stringify_table(msg)))
     end
   end
 
   self.on_error = function(uuid, err)
+    -- TODO: Implement a call to `getToken` here on certain failures, once I actually
```

Comment:
Once we know what an auth error actually looks like, we'll propagate it out of the Router to the Connection via this callback. This will allow us to handle propagating the oauth flow request to the firmware + marking the speaker offline. I should be able to figure this out tomorrow morning with the new guidance we got from devrel late today.